src/framework/extensions/internal/extensionsconfiguration.cpp

```diff
@@ -182,10 +162,64 @@ std::map<muse::Uri, Manifest::Config> ExtensionsConfiguration::manifestConfigs()
             }
         }
 
-        if (uri.isValid()) {
-            result[uri] = c;
-        }
+        return result;
     }
+    //! NOTE Load old plugins config
+    else {
+        ByteArray data;
+        mi::ReadResourceLockGuard lock_guard(multiInstancesProvider.get(), EXTENSIONS_RESOURCE_NAME);
+        Ret ret = io::File::readFile(oldPluginsConfigPath, data);
+        if (!ret) {
+            LOGE() << "failed read config data, err: " << ret.toString() << ", file: " << oldPluginsConfigPath;
+            return {};
+        }
 
-    return result;
+        std::string err;
+        JsonDocument doc = JsonDocument::fromJson(data, &err);
+        if (!err.empty()) {
+            LOGE() << "failed parse json, err: " << err;
+            return {};
+        }
+
+        if (!doc.isArray()) {
+            LOGE() << "bad format of config file";
+            return {};
+        }
+
+        std::map<std::string /*codeKey*/, Uri> uris;
+        {
+            legacy::ExtPluginsLoader pluginsLoader;
+            uris = pluginsLoader.loadCodekeyUriMap(pluginsDefaultPath(), pluginsUserPath());
+        }
+
+        std::map<Uri, Manifest::Config> result;
+        JsonArray arr = doc.rootArray();
+        for (size_t i = 0; i < arr.size(); ++i) {
+            JsonObject obj = arr.at(i).toObject();
+
+            Manifest::Config c;
+            std::string codeKey = obj.value("codeKey").toStdString();
+            Uri uri = muse::value(uris, codeKey);
+            if (!uri.isValid()) {
+                LOGW() << "not found plugin with codeKey: " << codeKey;
+                continue;
+            }
+
+            bool enabled = obj.value("enabled").toBool();
+            Action::Config ac;
+            ac.execPoint = enabled ? EXEC_MANUALLY : EXEC_DISABLED;
+            ac.shortcut = obj.value("shortcut").toStdString();
+
+            //! NOTE Special case
+            if (codeKey == "addCourtesyAccidentals") {
+                c.actions["add"] = ac;
+            } else {
+                c.actions["main"] = ac;
+            }
```

Comment:
What does this do and why is it necessary? 
And could you give a summary of how this concept of "Actions" works? Back when it was built, I didn't have time to look at all PRs, so I need to catch up a little bit :)

Reply:
It all started because of one plugin, or rather physically it was 3 plugins, but in essence it was one plugin of three parts - add, delete, configure

I didn't like that it looked like three plugins, each of which needs to be installed, enabled, each with a separate menu item..

so i decided to combine these three parts into one extension, which has three actions - that's where they appeared

```
{
    "uri": "musescore://extensions/courtesy_accidentals",
    "type": "composite",
    "title": "Courtesy Accidentals",
    "description": "This plugin adds courtesy accidentals",
    "category": "composing-arranging-tools",
    "thumbnail": "accidentals.png",
    "apiversion": 1,

    "actions": [
        {"code": "configure", "type": "form", "title": "Configure", "main": "configure.qml"},
        {"code": "add", "type": "macros", "title": "Add", "main": "add.js"},
        {"code": "remove", "type": "macros", "title": "Remove", "main": "remove.js"}
    ]
}
```
https://github.com/musescore/MuseScore/blob/master/share/extensions/courtesy_accidentals/manifest.json

Each action corresponds to its own menu item, now these are not three separate items, but one item with sub-items

![image](https://github.com/user-attachments/assets/fdc810e9-d038-4e83-b3fd-35a54ea40463)

If individual actions are not specified, then the default action is main.
```
{
    "uri": "musescore://extensions/colornotes",
    "type": "macros",
    "title": "Color Notes",
    "description": "This plugin colors notes in the selection depending on their pitch as per the Boomwhackers convention",
    "category": "color-notes",
    "thumbnail": "color_notes.png",
    "version":  "3.5",
    "apiversion": 1,

    "main": "main.js"
}
```

I plan to write documentation about all this :)

Reply:
Ah, I see. So if users want to do such things for their own plugins, they can do so, but will need to convert them to the new format. And this "special case" is only temporary until we convert out built-in plugins to the new format. Correct?